kafka-connect-http/src/main/java/com/github/castorm/kafka/connect/http/HttpSourceTask.java

```diff
@@ -141,6 +141,8 @@ public void commitRecord(SourceRecord record, RecordMetadata metadata) {
 
     @Override
     public void commit() {
+        if (null == confirmationWindow)
+            return;
```

Comment:
I'm missing the test you described in the PR.

On a separate note, I'm not a fan of null checks because of the (often) unnecessary complexity they add. What would you think of setting a default empty `ConfirmationWindow` on construction? That way this piece of logic would be unnecessary and the test would still pass, right?

Something like:
```java
private ConfirmationWindow<Map<String, ?>> confirmationWindow = new ConfirmationWindow<>(emptyMap());
```